sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/dependency.py

```diff
@@ -415,7 +540,91 @@ def get_upstream_dependency_inputs(
     """
     dependency_inputs = processing_input.ProcessingInputCollection()
     with db.Session() as session:
-        for dep in dependencies:
+        # -----------------------------
+        # Check for SPICE dependencies
+        # -----------------------------
+        # If spin is a dependency, query spin table for given date range
+        has_spin_dep = [dep for dep in dependencies if dep["data_source"] == "spin"]
+        if has_spin_dep:
```

Comment:
If you don't need to keep track of the items in the list you can use any() here instead to short-circuit the loop and not store references to the other items. Same with all of these conditionals.
```suggestion
        has_spin_dep = any(dep["data_source"] == "spin" for dep in dependencies)
        if has_spin_dep:
```